src/vga.js

```diff
@@ -385,253 +514,342 @@ VGAScreen.prototype.set_state = function(state)
     this.graphical_mode_is_linear = state[40];
     this.attribute_controller_index = state[41];
     this.offset_register = state[42];
+    this.planar_setreset = state[43];
+    this.planar_setreset_enable = state[44];
+    this.start_address_latched = state[45];
+    this.crtc.set(state[46]);
+    this.horizontal_display_enable_end = state[47];
+    this.horizontal_blank_start = state[48];
+    this.vertical_display_enable_end = state[49];
+    this.vertical_blank_start = state[50];
+    this.underline_location_register = state[51];
+    this.preset_row_scan = state[52];
+    this.offset_register = state[53];
+    this.palette_source = state[54];
+    this.attribute_mode = state[55];
+    this.color_plane_enable = state[56];
+    this.horizontal_panning = state[57];
+    this.color_select = state[58];
+    this.clocking_mode = state[59];
+    this.line_compare = state[60];
+    this.crtc_mode = state[61];
+    this.miscellaneous_graphics_register = state[62];
+    this.dac_state = state[63];
+    this.layers = state[64];
 
     this.bus.send("screen-set-mode", this.graphical_mode);
 
     if(this.graphical_mode)
     {
-        // TODO: Consider non-svga modes
-        this.set_size_graphical(this.svga_width, this.svga_height, this.svga_bpp);
+        if(this.svga_enabled)
+        {
+            this.set_size_graphical(this.svga_width, this.svga_height,
+                8, this.svga_width, this.svga_height);
+            this.update_layers();
```

Comment:
Does this restore the bpp correctly?

Reply:
Oops, nope it doesn't. I'll fix that up.